backend/src/api/model/block/mutations.rs

```diff
@@ -0,0 +1,355 @@
+        let updated_block = context.db(context.require_moderator()?)
+            .query_one(
+                &format!(
+                    "update blocks set
+                        title = case $2::boolean when true then $3 else title end
+                        where id = $1
+                        returning {}",
+                    Self::COL_NAMES,
+                ),
+                &[
+                    &id.key_for(Id::BLOCK_KIND)
+                        .ok_or_else(|| invalid_input!("`id` does not refer to a block"))?,
+                    &!set.title.is_implicit_null(),
+                    &set.title.some(),
+                ],
+            )
+            .await?;
+
+        Ok(Self::from_row(updated_block)?)
+    }
+
+    pub(crate) async fn update_text(id: Id, set: UpdateTextBlock, context: &Context) -> ApiResult<BlockValue> {
+        let updated_block = context.db(context.require_moderator()?)
+            .query_one(
+                &format!(
+                    "update blocks set
+                        title = case $2::boolean when true then $3 else title end,
+                        text_content = coalesce($4, title)
+                        where id = $1
+                        and type = 'text'
+                        returning {}",
+                    Self::COL_NAMES,
+                ),
+                &[
+                    &id.key_for(Id::BLOCK_KIND)
+                        .ok_or_else(|| invalid_input!("`id` does not refer to a block"))?,
+                    &!set.title.is_implicit_null(),
+                    &set.title.some(),
+                    &set.content,
+                ],
+            )
+            .await?;
+
+        Ok(Self::from_row(updated_block)?)
+    }
+
+    pub(crate) async fn update_series(id: Id, set: UpdateSeriesBlock, context: &Context) -> ApiResult<BlockValue> {
+        let updated_block = context.db(context.require_moderator()?)
+            .query_one(
+                &format!(
+                    "update blocks set
+                        title = case $2::boolean when true then $3 else title end,
+                        series_id = coalesce($4, series_id),
+                        videolist_layout = coalesce($5, videolist_layout),
+                        videolist_order = coalesce($6, videolist_order)
+                        where id = $1
+                        and type = 'series'
+                        returning {}",
+                    Self::COL_NAMES,
+                ),
+                &[
+                    &id.key_for(Id::BLOCK_KIND)
+                        .ok_or_else(|| invalid_input!("`id` does not refer to a block"))?,
+                    &!set.title.is_implicit_null(),
+                    &set.title.some(),
+                    &set.series.map(
+                        |series| series.key_for(Id::SERIES_KIND)
+                            .ok_or_else(|| invalid_input!("`set.series` does not refer to a series"))
```

Comment:
Yeah, again, I thought about something related, but decided against it, because it quickly proved to inflexible in my head. You can try coming up with something, though. :shrug:

Reply:
I added it to my own TODO list, lets see if I can come up with something.